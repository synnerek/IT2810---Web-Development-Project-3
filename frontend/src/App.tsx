import {Route, Routes} from 'react-router-dom';
import ProtectedRoutes from './ProtectedRoutes';
import HomePage from './Page/HomePage';
import DisplayMovie from './components/DisplayMovie';
import Header from './components/Header';
import Login from './components/Login';
import LikedMovies from './components/LikedMovies';
import CreateUser from './components/CreateUser';


function App() {
  return (
    <>
    <div>
      {sessionStorage.getItem("isLoggedIn") === "true" ? <Header/> : <></>}
        <Routes>
            <Route path='/login' element={<Login/>} />
            <Route path='/register' element={<CreateUser/>} />
            <Route element={<ProtectedRoutes />}> 
              <Route path="/" element={<HomePage/>} />
              <Route path="/movie/:movieID" element={<DisplayMovie />}/> 
              <Route path="/liked" element={<LikedMovies/>} />
          </Route>
        </Routes>
    </div>
    </>
  );
}
export default App;




