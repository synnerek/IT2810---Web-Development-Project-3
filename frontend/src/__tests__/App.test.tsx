import App from '../App';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { ApolloClient, ApolloProvider, InMemoryCache } from '@apollo/client';
import { RecoilRoot } from 'recoil';


test("First snapshot test", () => {

    const client = new ApolloClient({
        uri: 'http://localhost:3001/movie', 
        cache: new InMemoryCache(),
      });

    const component = renderer.create(
        <BrowserRouter>
            <ApolloProvider client={client}>
                <RecoilRoot>
                    <App />
                </RecoilRoot>
            </ApolloProvider>
        </BrowserRouter>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
});

export default {};
