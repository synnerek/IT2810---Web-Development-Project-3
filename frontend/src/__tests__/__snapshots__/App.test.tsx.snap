// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`First snapshot test 1`] = `
<div>
  <div
    className="href"
  >
    <div
      className="link"
    >
      <p>
        M O V I E W O R L D
      </p>
    </div>
  </div>
  <div
    className="MuiBox-root css-j7qwjs"
  >
    <h4
      className="MuiTypography-root MuiTypography-h4 css-141n43w-MuiTypography-root"
    >
      Login
    </h4>
    <div
      className="MuiFormControl-root MuiTextField-root css-kn4dux-MuiFormControl-root-MuiTextField-root"
      data-testid="username"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink={false}
        htmlFor=":r0:"
        id=":r0:-label"
      >
        Username
        <span
          aria-hidden={true}
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          disabled={false}
          id=":r0:"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="text"
          value=""
        />
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-yjsfm1"
          >
            <span>
              Username
               
              *
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      className="MuiFormControl-root MuiTextField-root css-kn4dux-MuiFormControl-root-MuiTextField-root"
      data-testid="password"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink={false}
        htmlFor=":r1:"
        id=":r1:-label"
      >
        Password
        <span
          aria-hidden={true}
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          disabled={false}
          id=":r1:"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="password"
          value=""
        />
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-yjsfm1"
          >
            <span>
              Password
               
              *
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-hxzusq-MuiButtonBase-root-MuiButton-root"
      data-testid="loginButton"
      disabled={true}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={-1}
      type="button"
    >
      login
    </button>
    <a
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-124h1b9-MuiButtonBase-root-MuiButton-root"
      href="/register"
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
    >
      Create new account
    </a>
  </div>
</div>
`;
